numba/targets/arrayobj.py

```diff
@@ -905,6 +905,58 @@ def array_argmax_impl(arry):
     return context.compile_internal(builder, array_argmax_impl, sig, args)
 
 
+@builtin
+@implement(numpy.median, types.Kind(types.Array))
+@implement("array.median", types.Kind(types.Array))
+def array_median(context, builder, sig, args):
+    def partition(A, low, high):
+        # median of three {low, middle, high}
+        middle = (low+high) // 2
+        if A[middle] > A[high]:
+            A[middle], A[high] = A[high], A[middle]
+        if A[low] > A[high]:
+            A[high], A[low] = A[low], A[high]
+        if A[middle] > A[low]:
+            A[low], A[middle] = A[middle], A[low]
+
+        # choose the middle as the pivot
+        A[high], A[middle] = A[middle], A[high]
+
+        x = A[high]
+        i = low-1
```

Comment:
I still think you can start with `i = low` if you move `i += 1` after the swap inside the loop.